internal/resources/diffValidation/resource_monitor.go

```diff
@@ -0,0 +1,68 @@
+//  (C) Copyright 2021 Hewlett Packard Enterprise Development LP
+
+package diffvalidation
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type LoadBalancerMonitor struct {
+	diff *schema.ResourceDiff
+}
+
+func NewLoadBalancerMonitorValidate(diff *schema.ResourceDiff) *LoadBalancerMonitor {
+	return &LoadBalancerMonitor{
+		diff: diff,
+	}
+}
+
+func (l *LoadBalancerMonitor) DiffValidate() error {
+	err := l.validateHttp()
+	if err != nil {
+		return err
+	}
+
+	err = l.validateHttps()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (l *LoadBalancerMonitor) validateHttp() error {
```

Comment:
not the right way to validate..

Reply:
what about icmp, udp, tcp cases ? Please re-think on this can be done in a better way.

Reply:
updated the proper validations for all types